src/hooks/useUserNameForm.ts

```diff
@@ -0,0 +1,39 @@
+import {FormEvent, useState} from "react";
+import {useNavigate} from "react-router-dom";
+
+const useUserNameForm = () => {
+    const [userName, setUserName] = useState<string>('');
+    const [userSurname, setUserSurname] = useState<string>('');
+    const MIN_LENGTH = 2
+
+    const navigate = useNavigate();
+    const validateNameSurname = (): boolean => {
+        return userName.length >= MIN_LENGTH && userSurname.length >= MIN_LENGTH; 
```

Comment:
one question, why do we have the validation separated in two different places and validating different things? shouldn't be easier to have all the validations in the same place?

Reply:
I renamed it for isLettersOnly. My idea here was that:

1 - The inputs should be text only, so I believe this validation should be immediate on typing. That's why I put it on the input's onChange property. This serves only to block unwanted characters.
2 - The name and surname validation refers to any wanted logic around if the name should be accepted for submission.

So one of the validations (isLettersOnly) is used to check if a typed character should be accepted or blocked. It's used on typing. If you try to type **;** (example) it won't appear.
The other (min text length) is used to check if a text should be submitted. It's used for submission and to enable the submit button.